FluentAvalonia/UI/Controls/IconElement/PathIcon.cs

```diff
@@ -25,41 +40,234 @@ public Geometry Data
+                case Stretch.Fill:
+                    if (double.IsInfinity(availableSize.Width))
+                    {
+                        sx = 1.0;
+                    }
+
+                    if (double.IsInfinity(availableSize.Height))
+                    {
+                        sy = 1.0;
+                    }
+
+                    break;
+                default:
+                    sx = sy = 1;
+                    break;
+            }
+            
+            // Apply stretch direction by bounding scales.
+            switch (stretchDirection)
+            {
+                case StretchDirection.UpOnly:
+                    if (sx < 1.0)
+                        sx = 1.0;
+                    if (sy < 1.0)
+                        sy = 1.0;
+                    break;
+
+                case StretchDirection.DownOnly:
+                    if (sx > 1.0)
+                        sx = 1.0;
+                    if (sy > 1.0)
+                        sy = 1.0;
+                    break;
+
+                case StretchDirection.Both:
+                    break;
+
+                default:
+                    break;
+            }
+
+            // Calculate translation in order to center the Icon
+            switch (stretch)
+            {
+                case Stretch.None:
+                    translate = Matrix.CreateTranslation(
+                        -shapeBounds.Position.X - (shapeBounds.Width - desiredX) / 2,
+                        -shapeBounds.Position.Y - (shapeBounds.Height - desiredY) / 2);
+                    break;
+
+                case Stretch.Uniform:
+                case Stretch.UniformToFill:
+                    if (sx != 0 && sy != 0)
+                    {
+                        translate = Matrix.CreateTranslation(
+                            -shapeBounds.Position.X - (shapeBounds.Width * sx - desiredX) / sx / 2,
+                            -shapeBounds.Position.Y - (shapeBounds.Height * sy - desiredY) / sy / 2);
+                    }
+                    else
+                    {
+                        translate = Matrix.CreateTranslation(-(Vector)shapeBounds.Position);
+                    }
+                    
+                    break;
+                case Stretch.Fill:
+                    translate = Matrix.CreateTranslation(-(Vector)shapeBounds.Position);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(Stretch), stretch, null);
+            }
+
+            var transform = translate * Matrix.CreateScale(sx, sy);
+            var size = new Size(shapeSize.Width * sx, shapeSize.Height * sy);
+            return (size, transform);
+        }
+
+        public override void Render(DrawingContext context)
+        {
+            var geometry = RenderedGeometry;
+            if (geometry == null)
+                return;
+
+            context.DrawGeometry(Foreground, null, geometry);
```

Comment:
This still needs the clip applied to it as in the old code. Or you can override the default value of `ClipToBounds` (to true) in the static ctor to auto handle this

Reply:
done with overriding in the static constructor, I find this better, as you can control it via style.